src/Bicep.Core/Emit/TemplateEmitter.cs

```diff
@@ -130,13 +129,19 @@ private EmitResult EmitOrFail(Func<SourceMap?> write)
             _ => [],
         };
 
+        var features = model switch
+        {
+            SemanticModel x => x.Features,
+            _ => null,
+        };
```

Comment:
Is this needed? I worry this might mislead people, because it's not really clear what it means to return "features" for a non-Bicep semantic model (e.g. an `ArmTemplateSemanticModel`)